yb-voyager/src/namereg/namereg_test.go

```diff
@@ -0,0 +1,636 @@
+		TargetName: newTableName(YUGABYTEDB, "public", "sakila", "table1"),
+	}
+	ntup.SetMode(IMPORT_TO_TARGET_MODE)
+
+	testCases := []struct {
+		pattern string
+		match   bool
+	}{
+		{"table1", false}, // effectively: <defaultSchema>.table1 i.e. public.table1
+		{"table2", false},
+		{"table", false},
+		{"TABLE1", true},
+		{"TABLE2", false},
+		{"TABLE", false},
+		{"TABLE*", true},
+		{"table*", false},
+		{"SAKILA.TABLE1", true},
+		{"SAKILA.TABLE2", false},
+		{"SAKILA.TABLE", false},
+		{"SAKILA.TABLE*", true},
+		{"SAKILA.table*", true}, // Schema name comparison is case insensitive. Matches with target name.
+		{"sakila.table1", true},
+		{"sakila.table2", false},
+		{"sakila.table", false},
+		{"sakila.table*", true},
+	}
+
+	for _, tc := range testCases {
+		match, err := ntup.MatchesPattern(tc.pattern)
+		assert.Nil(err)
+		assert.Equal(tc.match, match, "pattern: %s, expected: %b, got: %b", tc.pattern, tc.match, match)
+	}
+}
+
+//=====================================================
+
+var oracleToYBNameRegistry = &NameRegistry{
+	SourceDBType:              ORACLE,
+	mode:                      IMPORT_TO_TARGET_MODE,
+	SourceDBSchemaNames:       []string{"SAKILA"},
+	YBSchemaNames:             []string{"public"},
+	DefaultSourceDBSchemaName: "SAKILA",
+	DefaultYBSchemaName:       "public",
+	//DefaultSourceReplicaDBSchemaName: "SAKILA_FF", // Will be set using SetDefaultSourceReplicaDBSchemaName().
+	SourceDBTableNames: map[string][]string{
+		"SAKILA": {`TABLE1`, `TABLE2`, `Table2`, `MixedCaps`, `MixedCaps1`, `MixedCAPS1`, `lower_caps`},
+	},
+	YBTableNames: map[string][]string{
+		"public": {"table1", "table2", `Table2`, `mixedcaps`, `MixedCaps1`, `MixedCAPS1`, "lower_caps"},
+	},
+}
+
+func buildNameTuple(reg *NameRegistry, sourceSchema, sourceTable, targetSchema, targetTable string) *NameTuple {
+	ntup := &NameTuple{
+		SourceName: newTableName(reg.SourceDBType, sourceSchema, sourceSchema, sourceTable),
+		TargetName: newTableName(YUGABYTEDB, targetSchema, targetSchema, targetTable),
+	}
+	ntup.SetMode(reg.mode)
+	return ntup
+}
+
+func TestNameRegistrySuccessfulLookup(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	reg := oracleToYBNameRegistry
+	table1 := buildNameTuple(reg, "SAKILA", "TABLE1", "public", "table1")
+	table2 := buildNameTuple(reg, "SAKILA", "TABLE2", "public", "table2")
+	mixedCaps := buildNameTuple(reg, "SAKILA", "MixedCaps", "public", "mixedcaps")
+	lowerCaps := buildNameTuple(reg, "SAKILA", "lower_caps", "public", "lower_caps")
+
+	var testCases = []struct {
+		tableNames []string
+		expected   *NameTuple
+	}{
+		{[]string{
+			// YB side variants:
+			`table1`, `"table1"`, `public.table1`, `public."table1"`, `public."TABLE1"`, `public.TABLE1`,
+			// Oracle side variants:
+			`TABLE1`, `"TABLE1"`, `SAKILA.TABLE1`, `SAKILA."TABLE1"`, `SAKILA."table1"`, `SAKILA.table1`,
+		}, table1},
+		{[]string{"table2", "TABLE2"}, table2},
```

Comment:
If you add `TaBLe2` to this list, the test still passes. Do we want that? 
Basically, ORACLE DB has the following tables:
- TABLE2 (case insensitive)
- Table2 (case sensitive)

Now, if someone passes `TaBLe2` , is it safe to lookup and return the case insensitive match ? Maybe the user wanted `Table2` and made a mistake in specifying the case. Maybe we should fail here?

Reply:
```


casedb=# create table foo (k int primary key, v int);
CREATE TABLE
casedb=# create table "Foo" (k int primary key, v int);
CREATE TABLE
casedb=# \dt
             List of relations
 Schema |     Name      | Type  |  Owner
--------+---------------+-------+----------
 public | ALL_UPPERCASE | table | postgres
 public | Foo           | table | postgres
 public | MixedCase     | table | postgres
 public | all_lowercase | table | postgres
 public | foo           | table | postgres
(5 rows)

casedb=# select * from foo;
 k | v
---+---
(0 rows)

casedb=# select * from "Foo";
 k | v
---+---
(0 rows)

casedb=# select * from FOO;
 k | v
---+---
(0 rows)

```
Take a look at the above scenario: the db didn't say that `FOO` is ambiguous -- it picks the case insensitive variant. Whenever user needs precision they have to spell the exact case: with quotes on psql shell and without quotes on voyager cli.

The choice is also important in case of fall-back case:
Oracle: `SAKILA.TABLE1`, `SAKILA.Table1`
YB: `ybsakila.table1`, `ybsakila.Table1`
An event exported from YB for table `ybsakila.table1`. We __have__ to map the name to `SAKILA.TABLE1`.

Reply:
As discussed offline, 

if input to Lookup is quoted, we should force exact match. This does not happen in the current implementation. It falls back to the case insensitive match. 
- Lookup(fOo) should pick the case insensitive match. (foo)
- Lookup("Foo") should pick "Foo" 
- Lookup("fOo") should fail.